### PR TITLE
Update golang/x/net dependency on release-1.13

### PR DESCRIPTION
**Pull request by cblecker**
**What type of PR is this?**
/kind bug
/priority critical-urgent
/sig release network

**What this PR does / why we need it**:
Updates golang.org/x/net dependency to bring in http2 fix

**Which issue(s) this PR fixes**:
ref https://github.com/golang/go/issues/33606
ref #79912

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Update golang/x/net dependency to bring in fixes for CVE-2019-9512, CVE-2019-9514
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: hack/godep-save.sh [hunk 1]
```diff
@@ -52,9 +52,23 @@ if [[ -d Godeps ]]; then
     mv Godeps "${BACKUP}/Godeps"
 fi
 
+## Workaround "httplex" was dropped from golang.org/x/net repo and the code
+## was moved to the "golang.org/x/net/http/httpguts" directory, we do not use
+## this directly, however many packages we vendor are still using the older
+## golang.org/x/net and we need to keep this until all those dependencies
+## are switched to newer golang.org/x/net.
```
Comment (BenTheElder) [inline hack/godep-save.sh]: yikes

File: hack/godep-save.sh [hunk 2]
```diff
@@ -52,9 +52,23 @@ if [[ -d Godeps ]]; then
...
+REQUIRED_BINS=(
+  "golang.org/x/net/internal/nettest"
+  "golang.org/x/net/internal/socks"
+  "golang.org/x/net/internal/sockstest"
+)
+
 # Some things we want in godeps aren't code dependencies, so ./...
 # won't pick them up.
-REQUIRED_BINS=(
+REQUIRED_BINS+=(
```
Comment (BenTheElder) [inline hack/godep-save.sh]: tiny suggestion: invert this logic so the workaround follows and appends, so we change less in the future to undo it? /shrug

File: third_party/forked/godep/pkg.go [hunk 3]
```diff
@@ -42,9 +45,19 @@ func LoadPackages(names ...string) (a []*Package, err error) {
 	if len(names) == 0 {
 		return nil, nil
 	}
+
+	pkgs := strings.Split(ignorePackages, ",")
```
Comment (BenTheElder) [inline third_party/forked/godep/pkg.go]: nit / suggestion: can we call this variable something that makes it clear we want to ignore these?
`LoadPackages` containing a `pkgs` that is not what we're loading is slightly confusing imho.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve
two non-blocking nits / suggestions

**Comment by cblecker** [comment]
@BenTheElder https://github.com/kubernetes/kubernetes/pull/81546/commits/708702bf3e0d922bc764b1a744798b14d4fefb3e is a cherry pick of https://github.com/kubernetes/kubernetes/commit/eb4865fbe5e1b3ff987790d1d05fe2168e07b427 which has already been merged into master. I'd rather not make additional changes to it.

**Comment by cblecker** [comment]
All approvals are in, except for `third_party/`. As the code that will merge there is already a part of master, and we need this merged in order to cut a release, I'm going to use maintainer powers to manually stamp this as approved.

**Pull request merged**